pkg/tarmak/ssh/ssh.go

```diff
@@ -32,6 +33,34 @@ func New(tarmak interfaces.Tarmak) *SSH {
 	return s
 }
 
+func (s *SSH) Validate() error {
+	var result *multierror.Error
+
+	for _, path := range []string{
+		s.tarmak.Cluster().SSHConfigPath(),
+		s.tarmak.Environment().SSHPrivateKeyPath(),
+	} {
+
+		f, err := os.Stat(path)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+
+			result = multierror.Append(result, fmt.Errorf("failed to get '%s' file stat: %v", path, err))
+			continue
+		}
+
+		if f.Mode() != os.FileMode(0600) {
+			err := fmt.Errorf("'%s' does not match permissions (0600): %v", path, f.Mode())
+			result = multierror.Append(result, err)
+			continue
+		}
```

Comment:
I think we should be only checking if others and group are having a zero like that

`(f.Mode() & 0077) == 0`